src/util/PropertyHelper.js

```diff
@@ -0,0 +1,25 @@
+import { camelToDash } from './AttributeParser';
+
+/**
+ * Helper Function to inform to call the callbacks for  watched properties and dispatch change events
+ *
+ * @param context is the elements instance
+ * @param property property that changes
+ * @param newValue
+ * @param oldValue
+ */
+export const informWatchedPropertiesAndDispatchChangeEvent = (context, property, newValue, oldValue) => {
```

Comment:
@quarkus I don't like that this assumes a few things on the context. In this case it "knows" that the context has a watch method and propertyOptions and so on... But why does it know that?! 
Would it be so wrong to duplicate this small piece of code? I wouldn't mind.

Reply:
Duplicating that piece of code what open the door to potential runtime differences differences for the exact same thing.

If one implementation is changed in the future and the other one is forgotten this could lead to super weird and hard to track behaviour.

Thats why i extracted the method in the first place.

The only alternative or me would be to simply ignore the dispatch part in the store update and just call the watcher callbacks.